craft_providers/lxd/launcher.py

```diff
@@ -72,48 +105,96 @@ def _create_instance(
-            "Creating new base instance %r from instance.", base_instance.instance_name
+            "Creating new base instance from image %r from remote %r",
+            image_name,
+            image_remote,
+        )
+        base_instance.launch(
+            image=image_name,
+            image_remote=image_remote,
+            ephemeral=False,  # base instance should not ephemeral
+        )
+        base_instance_status = base_instance.config_get("user.craft_providers.status")
+
+        # Skip the base configuration if the instance is already configured.
+        if base_instance_status != "FINISHED":
+            logger.debug("Setting up base instance %r", base_instance.instance_name)
+            base_instance.config_set("user.craft_providers.status", "PREPARING")
+            config_timer = InstanceTimer(base_instance)
+            config_timer.start()
+            base_configuration.setup(executor=base_instance)
+            _set_timezone(
+                base_instance,
+                base_instance.project,
+                base_instance.remote,
+                base_instance.lxc,
+            )
+            base_instance.config_set("user.craft_providers.status", "FINISHED")
+            # set the full instance name as image description
+            lxc.config_set(
+                instance_name=base_instance.instance_name,
+                key="image.description",
+                value=base_instance.name,
+                project=project,
+                remote=remote,
+            )
+            config_timer.stop()
+            base_instance.stop()
+
+        # Copy the base instance to the instance.
+        logger.info("Creating new instance from base instance")
+        logger.debug(
+            "Creating new instance %r from base instance %r",
+            instance.instance_name,
+            base_instance.instance_name,
         )
         lxc.copy(
-            source_remote=remote,
-            source_instance_name=instance.instance_name,
+            source_remote=base_instance.remote,
+            source_instance_name=base_instance.instance_name,
             destination_remote=remote,
-            destination_instance_name=base_instance.instance_name,
+            destination_instance_name=instance.instance_name,
             project=project,
         )
-
-        # set the full instance name as image description
-        lxc.config_set(
-            instance_name=base_instance.instance_name,
-            key="image.description",
-            value=base_instance.name,
-            project=project,
-            remote=remote,
+        _set_timezone(instance, project, remote, lxc)
+    else:
+        instance.launch(
+            image=image_name, image_remote=image_remote, ephemeral=ephemeral
         )
+        instance_status = instance.config_get("user.craft_providers.status")
+
+        # Skip the base configuration if the instance is already configured.
+        if instance_status != "FINISHED":
```

Comment:
What scenario could occur for the status to be `FINISHED` here?

Reply:
It could have configuration error or network issues, which the base instance image can be used but stopped at preparing. just need to redo the configuration part.